Add a blog article about leaf nodes

smortex opened the pull request.

ripienaar commented on blog/content/post/2021/03/23/foo.md:
```diff
@@ -0,0 +1,90 @@
+author: Romain Tartière
+---
+
+Because Choria allows you to manage nodes spread all around the world, and because you might be working from your laptop, far away from the (bad) Wi-Fi access points that connects you through (bad) PLC to the (bad) internet connexion from the (not bad) island you are on, you may experience inconvenient lattency and unreliabilities.
```
`connection`, few other places too

ripienaar commented on blog/content/post/2021/03/23/foo.md:
```diff
@@ -0,0 +1,90 @@
+
+It is now possible to use the `choria-choria` puppet module to setup leaf nodes, this section shows how to proceed.
+
+Before starting, you might want to measure the lattency when you start to compare with what you have at the end.  Let's use `choria ping` for this purpose:
```
`latency`

ripienaar commented on blog/content/post/2021/03/23/foo.md:
```diff
@@ -0,0 +1,90 @@
+
+All numbers are expressed in milliseconds.
+
+The _max_ / _min_ / _avg_ figures are not the one we are interested in here because they measure the lattency between the choria broker and the choria servers.  The _overhead_ is what we want to improve, nearly 2 seconds in my case.
```
latency between choria client and choria servers via the broker once already connected

smortex replied:
Not 100% sure if it is accurate, I added this clarification which would need proofreading:

> The _overhead_ is what we want to improve: _the time needed by the client to establish a connexion to the broker node_, nearly 2 seconds in my case.

ripienaar replied:
Yes close enough - the other thing it includes is the security layer time, in my case my AAA server is in the netherlands and that round trip is also included in overhead.  You runnin tradditional cert based access dont have much measuarable time there (micro seconds)

ripienaar replied:
so for all intents your description is correct :)

ripienaar commented on blog/content/post/2021/03/23/foo.md:
```diff
@@ -0,0 +1,90 @@
+  choria2:
+    url: "nats://choria2.example.net:7422"
+  choria3:
+    url: "nats://choria3.example.net:7422"
```
assuming this is one remote choria network you should do `url: nats://choria1:7422,nats://choria2:7422` rather than 3 connections, this works better with nats internals and 1 connection is plenty

smortex replied:
All my setups have a single broker, so I have basically no experience with this kind of setup and have not read much on the topic.

Are there use cases for which splitting the entries would be better?

ripienaar replied:
In real nats yes, you can connect to different seperate clusters into different seperate authentication domains (like VLANs) and from the leafnode - using an entirely different auth schema or even no TLS - access thoe networks.

While I support all that in choria its not exposed in the modules so for us, a basic 1 leaf is probably what makes sense now

ripienaar commented on blog/content/post/2021/03/23/foo.md:
````diff
@@ -0,0 +1,90 @@
+47 replies max: 509.744 min: 171.929 avg: 318.926 overhead: 47.991
+```
+
+The _overhead_ has reduced from 1940.673 to 47.991 ms, about 2.5% of the initial duration.
````
looks pretty great so far, thank you :)

smortex commented on blog/content/post/2021/03/23/foo.md:
```diff
@@ -0,0 +1,85 @@
+---
+title: "Running Choria Across the Globe"
+title: "Fighting Latency with Choria"
```
Maybe "Fighting Latency" is probably not the best title, since the actual latency increase slightly with this setup… "Reducing overhead", "Faster connections" or something like that might be better or maybe something more "technical" like "Running Choria Leaf Nodes", I don't know what is better :-/

ripienaar replied:
`Reducing connection overhead for branch office scenarios` or something like that, thats probably too long - i always struggle with titles too

smortex replied:
Well, a long and accurate title is better than a short one that is very vague :-)

smortex commented:
I adjusted a few more things and added an image.  Once the new release is done, I will have to adjust the link to it and i think we are not far away from something acceptable.

A proper title is needed though :-)

ripienaar commented:
Will take a close look again in the morning :)

ripienaar commented:
So I shipped latest modules for versions information and you can update the dates and such to publish, if you do this edit night my time (central europe) set it for the next day and I'll merge and release on schedule

thanks!

smortex commented:
I updated the links and title, it should be okay.  I assumed we will release 0.21.1 to fix the leaf node account issue, so said that this is the first version supporting this.

While here, I also updated the formatting of `choria ping`.

ripienaar commented:
Yeah, I think I will try for this weekend to do a .1

ripienaar commented:
We could now ressurect this blog post, releases are out and things should work :) you'd need to move it to a new date but I think we're pretty much good to ship this

smortex commented:
I updated the links and the date to be next Monday.  This seems to be working as expected and we can get this merged :+1:
